src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -337,6 +337,24 @@ MTR_AVAILABLE(ios(16.1), macos(13.0), watchos(9.1), tvos(16.1))
                     queue:(dispatch_queue_t)queue
                completion:(void (^)(NSURL * _Nullable url, NSError * _Nullable error))completion
     MTR_AVAILABLE(ios(17.6), macos(14.6), watchos(10.6), tvos(17.6));
+
+/**
+ * Wait until a set of attributes reaches certain values or the provided timeout
+ * (if any) is reached.  The completion is called at that point, with an error
+ * if a timeout occurred.
+ *
+ * The completion is also called with an error on input errors.
+ *
+ * The values to wait for are represented as an array of response-values which
```

Comment:
Is that really a convenient representation for clients here? The whole thing could be a dictionary from attribute paths to values for example.

Reply:
Hmm, that's a good point.  Will change that.

Reply:
It's a bit of a tough call, in that the format here is what we actually report attributes as to consumers... But I agree that a typical use of this would probably do better with the paths-to-values dictionary.